samples/typescript_nodejs/13.core-bot/src/dialogs/flightBookingRecognizer.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { RecognizerResult, TurnContext } from 'botbuilder';
+import { LuisApplication, LuisRecognizer } from 'botbuilder-ai';
+
+export class FlightBookingRecognizer {
+    private recognizer: LuisRecognizer;
+
+    constructor(config: LuisApplication) {
+        const luisIsConfigured = config && config.applicationId && config.endpoint && config.endpointKey;
+        if (luisIsConfigured) {
+            this.recognizer = new LuisRecognizer(config, {}, true);
+        }
+    }
+
+    public get isConfigured(): boolean {
+        return (this.recognizer !== undefined);
+    }
+
+    /**
+     * Returns an object with preformatted LUIS results for the bot's dialogs to consume.
+     * @param {TurnContext} context
+     */
+    public async executeLuisQuery(context: TurnContext): Promise<RecognizerResult> {
+        return await this.recognizer.recognize(context);
+    }
+
+    public getFromEntities(result) {
+        let fromValue, fromAirportValue;
+        if (result.entities.$instance.From) {
+            fromValue = result.entities.$instance.From[0].text;
+        }
+        if (fromValue && result.entities.From[0].Airport) {
+            fromAirportValue = result.entities.From[0].Airport[0][0];
+        }
+
+        return { from: fromValue, airport: fromAirportValue };
+    }
+
+    public getToEntities(result) {
+        let toValue, toAirportValue;
+        if (result.entities.$instance.To) {
```

Comment:
Can the `To` property ever be empty? If so, it might be possible that a result with the following shape:

```
result: {
  entities: {
    $instance: {
      To: []
    }
  }
}
```
passes this `if` check, but then the below reference to `result.entities.$instance.To[0].text` throws an error because there is no first element at index 0.

If this is possible, you might want to check `.To.length` as well.

Reply:
@gabog can you chime in on this?

Reply:
Resolving per https://github.com/microsoft/BotBuilder-Samples/pull/1648#discussion_r303625530